android/tangram/src/com/mapzen/tangram/FontFileParser.java

```diff
@@ -0,0 +1,138 @@
+
+    private void processDocument(XmlPullParser parser) throws XmlPullParserException, IOException {
+
+        parser.nextTag();
+        // Parse Families
+        parser.require(XmlPullParser.START_TAG, null, "familyset");
+        while (parser.next() != XmlPullParser.END_DOCUMENT) {
+            if (parser.getEventType() != XmlPullParser.START_TAG) {
+                continue;
+            }
+            if ("family".equals(parser.getName())) {
+                // Parse this family:
+                String name = parser.getAttributeValue(null, "name");
+                while (parser.next() != XmlPullParser.END_TAG) {
+                    if (parser.getEventType() != XmlPullParser.START_TAG) {
+                        continue;
+                    }
+                    String tag = parser.getName();
+                    if ("font".equals(tag)) {
+                        String weightStr = parser.getAttributeValue(null, "weight");
+                        weightStr = (weightStr == null) ? "400" : weightStr;
+
+                        String styleStr = parser.getAttributeValue(null, "style");
+                        styleStr = (styleStr == null) ? "normal" : styleStr;
+
+                        String filename = parser.nextText();
+                        String fullFilename = "/system/fonts/" + filename;
+
+                        String key = name + "_" + weightStr + "_" + styleStr;
+                        fontDict.put(key, fullFilename);
+                    } else {
+                        skip(parser);
+                    }
+                }
+            } else if ("alias".equals(parser.getName())) {
+                // Parse this alias to font to fileName
+                String aliasName = parser.getAttributeValue(null, "name");
+                String toName = parser.getAttributeValue(null, "to");
+                String weightStr = parser.getAttributeValue(null, "weight");
+                weightStr = (weightStr == null) ? "400" : weightStr;
+                String fontFilename = fontDict.get(toName + "_" + weightStr + "_normal"); // alias style is default: normal (always)
+                fontDict.put(aliasName + "_" + weightStr, fontFilename);
+            } else {
+                skip(parser);
+            }
+        }
+    }
+
+    private void skip(XmlPullParser parser) throws XmlPullParserException, IOException {
+        int depth = 1;
+        while (depth > 0) {
+            switch (parser.next()) {
+            case XmlPullParser.START_TAG:
+                depth++;
+                break;
+            case XmlPullParser.END_TAG:
+                depth--;
+                break;
+            }
+        }
+    }
+
+    public FontFileParser() {
+
+        InputStream in = null;
+        final File fontFile = new File("/system/etc/fonts.xml");
+        final File systemFontFile = new File("/system/etc/system_fonts.xml");
+        final File fallbackFontFile = new File("/system/etc/fallback_fonts.xml");
```

Comment:
The three files named here all have different layouts (on my phone, at least). We can't really treat them as substitutes because `parseDocument` will simply fail on a file that doesn't have the same layout as fonts.xml